// katacoda-cli/lib/helpers/constants.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.defaultImages = {
    "Other": "other",
    "Ubuntu 18.04": "ubuntu:1804",
    "Ubuntu 20.04": "ubuntu:2004",
    "Kubernetes Cluster 1.18": "kubernetes-cluster:1.18",
    "Kubernetes Cluster 1.18 (Pre-configured)": "kubernetes-cluster-running:1.18",
    "Java / Scala": "openjdk:15",
    "Kotlin": "kotlin:1.3.61",
    "Python 3.6": "python:3.6",
    "Python 3.7": "python:3.7",
    "Python 3.8": "python:3.8",
    "Go": "golang:1.14",
    "Rust": "rust:1.44.1",
    "Ruby": "ruby:2.5",
    "C# (dotnet core 3.1)": "dotnet:3.1",
    "R": "rlang:3.4",
    "Node.js": "nodejs:12",
    "Bash": "bash"
};
exports.deprecatedImages = {
    "ubuntu": "Ubuntu 16.04",
    "ubuntu:1604": "Ubuntu 1604",
    "ubuntu1804": "Ubuntu 18.04",
    "ubuntu1904": "Ubuntu 19.04",
    "kubernetes-cluster": "Kubernetes Cluster (Kubeadm, 2 Node Cluster)",
    "docker": "Docker",
    "kubernetes-cluster-running": "Kubernetes Cluster Running (Kubeadm, 2 Node Cluster, Already Configured)",
    "kubernetes-cluster-running:1.14": "Kubeadm 1.14 2 Node Cluster Running"
};
// These values are used as default if the CLI can't retrieve the values through the API
// editor.katacoda.com/validation/indexjson/layout
exports.defaultLayouts = {
    Terminal: 'terminal',
    '2x Terminal Split': 'terminal-terminal',
    'Editor + Terminal': 'editor-terminal',
    'Terminal + Editor for the embedded presentation': 'editor-terminal-split',
    'Terminal + Iframe': 'terminal-iframe'
};
exports.defaultDifficultyLevels = [
    'Beginner',
    'Intermediate',
    'Advanced'
];
exports.defaultLayoutsImages = {
    'Docker Swarm': 'terminal-terminal',
    'Kubernetes Cluster': 'terminal-terminal'
};
